dask/array/core.py

```diff
@@ -3384,8 +3384,10 @@ def to_zarr(
 
     if isinstance(url, zarr.Array):
         z = url
-        if isinstance(z.store, (dict, MutableMapping)) and "distributed" in config.get(
-            "scheduler", ""
+        if (
+            isinstance(z.store, (dict, MutableMapping))
+            and not callable(config.get("scheduler", ""))
+            and "distributed" in config.get("scheduler", "")
```

Comment:
I think this can be a bit broader, to allow for scheduler to be a class as well.

```suggestion
            and isinstance(config.get("scheduler", None), str)
            and "distributed" in config.get("scheduler", "")
```

Reply:
Hmm I think @TnTo wants the `ray_on_dask` scheduler to also raise an error here, similar to the distributed scheduler. @TnTo can you provide more details around the changes here?

Reply:
Another option would be to allow `compute_kwargs` on this method. It looks like these would just need to be passed into the `arr.store` method here: https://github.com/dask/dask/blob/0d4bdea8d5215270646c7b48413e4125f590166e/dask/array/core.py#L3451

Reply:
> Hmm I think @TnTo wants the `ray_on_dask` scheduler to also raise an error here, similar to the distributed scheduler. @TnTo can you provide more details around the changes here? 

Oh did I invert the logic? Yeah a test would definitely help with my understanding of the intended behavior.

Reply:
You inverted the logic, but don't worry :smile: 

As it is implemented now `to_zarr` wants that `dask.config["scheduler"]` is an iterable because of the `in` check. I imagine that it is a good behaviour if only dask internal scheduler are considered because they are represented by strings.

This prevent the function to be used with custom scheduler like ray. For example, (ray docs)[https://docs.ray.io/en/latest/data/dask-on-ray.html#scheduler] suggests to set global scheduler with `dask.config.set(scheduler=ray_dask_get)` where `ray_dask_get` is a function and so not an iterable.

The intention of the commit is to allow `to_zarr` to work with external scheduler and not only internal one. I think that if the ray scheduler would be incompatible with `to_zarr` (at least on a single machine it is not) the issue would be better addressed by ray itself.

In `base.py`, line 1250 a very similar check is written as `scheduler in ("dask.distributed", "distributed")`; so if the callable check is too ad hoc an alternative could be to rewrite the check as 
`        if (
            isinstance(z.store, (dict, MutableMapping))
            and config.get("scheduler", "") in ("dask.distributed", "distributed")
        ):
`
to explicitly excluding only the cases of interest.

I hoper to have clarify a little.

Reply:
I agree that `config.get("scheduler") in ("dask.distributed", "distributed")` seems like the safest check here.